app/src/main/java/com/example/myforecast/viewmodels/WeatherInfoViewModel.kt

```diff
@@ -0,0 +1,36 @@
+package com.example.myforecast.viewmodels
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.domain.entities.WeatherInformation
+import com.example.domain.usecases.GetDailyWeatherByLatitudeAndLongitudeUseCase
+import com.example.domain.util.Result
+import com.example.myforecast.utils.Event
+import com.example.myforecast.utils.DataStatus
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class WeatherInfoViewModel(val getDailyWeatherByLatitudeAndLongitude: GetDailyWeatherByLatitudeAndLongitudeUseCase)
+    : ViewModel() {
+
+    private var _mainState: MutableLiveData<Event<DataStatus<WeatherInformation>>> = MutableLiveData()
+    val mainState: LiveData<Event<DataStatus<WeatherInformation>>>
+        get() {
+            return _mainState
+        }
+
+    fun onRemoteSearch() = viewModelScope.launch {
+        _mainState.value = Event(DataStatus.Loading)
+        when (val result = withContext(Dispatchers.IO) { getDailyWeatherByLatitudeAndLongitude(getFromRemote = true) }) {
+            is Result.Failure -> {
+                _mainState.value = Event(DataStatus.Error(error = result.exception))
```

Comment:
Por què todo se maneja còmo un evento? Aquì hay que saber el porque se utiliza el wrapper, sin embargo no significa que siempre se envìe la informaciòn asì.

Reply:
El proyecto guía original en el cual esta basado tambien lo define, con el fin de usar el wrapper para poder manejar la información del liveData que debe ser usada una sola vez (_https://medium.com/androiddevelopers/livedata-with-snackbar-navigation-and-other-events-the-singleliveevent-case-ac2622673150_). Sin embargo no tiene uso inmediato en este proyecto y revisando el proyecto guía tampoco tiene uso en ese.

Reply:
Està bien seguir los ejemplos, el detalle aquì es que el event puede ser para un caso y no para todos. Estarìa bien aplicar sòlo los requeridos, ya sea que el ejemplo lo tenga, pero si se analiza se puede identificar que hay casos que si aplican y otros que no, en este mismo proyecto.